Swope brandi install updated (wtg on developer) (#6968)
open-cluster-management/rhacm-docs: install/install_connected.adoc

[#installing-while-connected-online]
= Installing while connected online

> Install {acm} through {olm}, which manages the installation, upgrade, and removal of the components that encompass the {acm-short} hub cluster. When you install the {acm-short} operator, you create the `MultiClusterHub` resource. The `MultiClusterHub` resource then installs the {mce} and creates the `MultiClusterEngine` resource. You must have a supported version of {ocp-short} to install {acm-short}.

Author: This is the explanation that we need to tell the user how MCE is installed with the MCH resource. @cameronmwall

Reviewer: I actually think we don't need to go into details on how the MCE is installed. All of that is happening under the covers and the specific mechanisms are not relevant to most of our customers

Author: I get that, but we get questions from people using OCP and not ACM, so we are attempting to explain it for others using the doc. I would like explain it a bit better here.

Reviewer 2: @swopebe I think this is where you were looking for input while Cameron is OOTO.  I understand your point that customers are asking about the MCE operator/MCE resource tht automatically appear when they install ACM, so it is fine to have info here that says that is going to happen.     Maybe it would be helpful to add a little "why?" rationale here, namely stating that ACM depends on and uses MCE.   So perhaps something like this (please edit/fix the terminology/placeholders)...

Because Advanced Cluster Management for Kubernetes  depends on and uses multicluster engine, after you create the `MultiClusterHub` resource, the {acm-short} operator automatically installs the {mce} operator and creates the `MultiClusterEngine` resource. 

You must have a supported version of {ocp-short} to install {acm-short}.

> Install {acm} through {olm}, which manages the installation, upgrade, and removal of the components that encompass the {acm-short} hub cluster. When you install the {acm-short} operator, you create the `MultiClusterHub` resource. The `MultiClusterHub` resource then installs the {mce} and creates the `MultiClusterEngine` resource. You must have a supported version of {ocp-short} to install {acm-short}.

Reviewer 3: Strong description !

Author: updated to match the overview

> **Required access:** Cluster administrator

Reviewer 3: ```suggestion
*Required access:* Cluster administrator
```

> **{ocp-short} Dedicated environment required access:** You need `cluster-admin` permissions. By default the `dedicated-admin` role does not have the required permissions to create namespaces in the {ocp-short} Dedicated environment. 

Reviewer 3: ```suggestion
*{ocp-short} Dedicated environment required access:* You need `cluster-admin` permissions. By default the `dedicated-admin` role does not have the required permissions to create namespaces in the {ocp-short} Dedicated environment. 
```

*Notes:*

> - By default, the hub cluster components are installed on worker nodes of your {ocp-short} cluster without any additional configuration. You can install the hub cluster on worker nodes by using the {ocp-short} OperatorHub web console interface, or by using the {ocp-short} CLI.

Reviewer 3: Should there be a space in OperatorHub? Should it be Operator Hub ?

Author: I will have to look. I didn't write it, so I didn't think to research that.

Author: I will have to look. I didn't write it, so I didn't think to research that.

Author: We are correct in the naming: https://docs.redhat.com/en/documentation/openshift_container_platform/4.14/html/operators/understanding-operators

- If you have configured your {ocp-short} cluster with infrastructure nodes, you can install the hub cluster on those infrastructure nodes by using the {ocp-short} CLI with additional resource parameters. See the _Installing the {acm-short} hub cluster on infrastructure node_ section for more details.

- If you plan to import Kubernetes clusters that are not {ocp-short} or {acm-short} clusters, you need to configure an image pull secret. 

> 

Author: present tense

[#confirm-ocp-installation]
== Confirm your {ocp-short} installation

> Verify that a {product-title-short} hub cluster is not already installed on your {ocp-short} cluster.You cannot have more than one hub cluster. 

Reviewer 4: ```suggestion
Verify that a {product-title-short} hub cluster is not already installed on your {ocp-short} cluster. You cannot have more than one hub cluster. 
```
Small typo (missing space)

> Verify that a {product-title-short} hub cluster is not already installed on your {ocp-short} cluster.You cannot have more than one hub cluster. 

Reviewer 3: Reminder that the conref changed for the product name. The title of this section should also be changed to "Confirming"
```suggestion
Verify that a {acm-short} hub cluster is not already installed on your {ocp-short} cluster. You cannot have more than one hub cluster. 
```

> {acm-short} allows only one single {acm-short} hub cluster installation on each {ocp-short} cluster. Continue with the following steps if there is no {acm-short} hub cluster installed:

Reviewer 3: ```suggestion
You can only have one single {acm-short} hub cluster installation on each {ocp-short} cluster. Continue with the following steps if there is no {acm-short} hub cluster installed:
```

Author: This one is OK either way, using **have** too much is also frowned upon. My only change here was taking out the redundant line before it. This file had too much content.

. To ensure that the {ocp-short} cluster is set up correctly, access the {ocp-short} web console with the following command:

+
----

> oc -n openshift-console get route

Reviewer 3: Reminder to add the [source,bash] syntax for this command and any others that might be missing it

Author: @dockerymick much of the file is missing all of that--across the book. I didn't clean up all those instances due to time. I can change this one but it is not the only one. I only made the command correct.

> ----

Author: obvious change here

[#confirm-ocp-installation]
== Confirm your {ocp-short} installation

> Verify that a {product-title-short} hub cluster is not already installed on your {ocp-short} cluster. You cannot have more than one hub cluster. 

Reviewer 3: ```suggestion
Verify that a {acm-short} hub cluster is not already installed on your {ocp-short} cluster. You cannot have more than one hub cluster. 
```

. Open the URL in your browser and check the result. If the console URL displays `console-openshift-console.router.default.svc.cluster.local`, set the value for `openshift_master_default_subdomain` when you install {ocp-short}. See the following example of a URL: `https://console-openshift-console.apps.new-coral.purple-chesterfield.com`.

You can proceed to install {acm-short} from the console or the CLI. 

* <<installing-from-the-operatorhub,Installing from the OperatorHub web console interface>>

> * <<installing-from-the-cli,Installing from the {ocp-short} CLI>>

Reviewer 3: These child links can be removed since you already list them in the TOC on lines 20-24

Author: These were already there, the only change I had was to remove the extra line that is obvious.

Author: Given the length of this file, I don't mind it and I noticed that Alexandra does this, as well as other team. I think this can stay because it can help orient the user in a topic that has a lot going on.

Reviewer 3: Ok, I think this is something worth bringing up in a future doc arch call

[#installing-from-the-operatorhub]
== Installing from the OperatorHub web console interface

> **Best practice:** From the _Administrator_ view in your console, install the OperatorHub web console interface that is provided with {ocp-short}.

Reviewer 3: ```suggestion
*Best practice:* From the _Administrator_ view in your console, install the OperatorHub web console interface that is provided with {ocp-short}.
```

. Select *Operators* > *OperatorHub* to access the list of available operators, and select _Advanced Cluster Management for Kubernetes_ operator.

. On the _Operator subscription_ page, select the options for your installation:

Channel:: The channel that you select corresponds to the release that you are installing. When you select the channel, it installs the identified release, and establishes that the future Errata updates within that release are obtained.

Namespace information:: The {acm-short} hub cluster must be installed in its own namespace, or project. 

> - By default, the OperatorHub console installation process creates a namespace that is titled `open-cluster-management`. *Best practice:* Continue to use the `open-cluster-management` namespace if it is available.  

Author: The important thing here is that they keep it `open-cluster-management` and I think that is not as clear now. The second way makes it seem to me that it is a _general_ name, but it is a specific name. (would use "an" here)

hmmm, I think we need to word "titled" here for that clarity??

  
- If there is already a namespace named `open-cluster-management`, choose a different namespace.

Approval strategy for updates:: The approval strategy identifies the human interaction that is required for applying updates to the channel or release to which you subscribed. 

- Select *Automatic* to ensure any updates within that release are automatically applied. 
  
- Select *Manual* to receive a notification when an update is available. If you have concerns about when the updates are applied, this might be best practice for you.
+
*Important:* To upgrade to the next minor release, you must return to the _OperatorHub_ page and select a new channel for a more recent release.

> . Select *Install* to apply your changes and create the operator. 

Reviewer 3: MultiClusterHub on line 108, should be monospaced because it's introduced as a resource name

Author: @dockerymick since it is UI Control, and already bolded, I am not sure we would do that here. (?) When we bold something in the UI, it is to make it look _exactly_ like the UI, at least as much as possible, so the button itself I don't think has it monospaced. Not sure I have even asked this question to myself. 

In the UI, I think (would have to confirm another time) that it looks just like this: **MultiClusterHub**

Author: @dockerymick since it is UI Control, and already bolded, I am not sure we would do that here. (?) When we bold something in the UI, it is to make it look _exactly_ like the UI, at least as much as possible, so the button itself I don't think has it monospaced. Not sure I have even asked this question to myself. 

In the UI, I think (would have to confirm another time) that it looks just like this: **MultiClusterHub** and the other way we are saying it would look like _**MultiClusterHub**_. Good observation; not sure.

 .. Select the *MultiClusterHub* tab.
 .. Select *Create MultiClusterHub*.
 .. Update the default values in the YAML file. See options in the _MultiClusterHub advanced configuration_ section of the documentation.

> . Click to find that your operator is listed.

Author: @cameronmwall we had a note to add a verification step during our doc test, but I forgot the thing they need to click and I didn't write it down.

Can you get me a better verification step here for them to see the MCH resource?

Reviewer: so, if we want for them to be able to be sure that their installation was successful I think we can say something along the lines of "clicking the 'MultiClusterHub' tab will show a list of resources which includes the MCH resource just created. When the status becomes 'Running', the installation is complete"

 .. Select the *MultiClusterHub* tab.
 .. Select *Create MultiClusterHub*.
 .. Update the default values in the YAML file. See options in the _MultiClusterHub advanced configuration_ section of the documentation.

> . Click to `MultiClusterHub` tab to see the list of resources where your operator is listed.

Reviewer 3: This should be italicized since it is the name of tab in the console
```suggestion
. Click to _MultiClusterHub_ tab to see the list of resources where your operator is listed.
```

> . Click to _MultiClusterHub_ tab to see the list of resources where your operator is listed.

Reviewer 3: ```suggestion
. Click the _MultiClusterHub_ tab to see the list of resources where your operator is listed.
```

----

+ 

> . Select *Create* to initialize the custom resource. It can take up to 10 minutes for the {product-title-short} hub cluster to build and deploy components.

Reviewer 3: Reminder that the conref for the product name was changed to acm and acm-short
```suggestion
. Select *Create* to initialize the custom resource. It can take up to 10 minutes for the {acm-short} hub cluster to build and deploy components.
```

+

> After the {product-title-short} hub cluster is created, the `MultiClusterHub` resource status displays _Running_ from the _MultiClusterHub_ tab of the {product-title-short} operator details. 

Reviewer 3: ```suggestion
After the {acm-short} hub cluster is created, the `MultiClusterHub` resource status displays _Running_ from the _MultiClusterHub_ tab of the {acm-short} operator details. 
```

> To gain access to the console, see the link:../console/console_access.adoc#accessing-your-console[Accessing your console] topic.

Reviewer 3: I can understand why this link is added, but I recommend moving it to the Additional resources section. Let me know your thoughts

Author: I didn't add the link, only separated it out...it could be a follow on step even, because at this point the user goes to the UI. I will add it.

[#installing-from-the-cli]
== Installing from the {ocp-short} CLI

> Install the operator and the objects.

Reviewer 3: ```suggestion
Install the operator and the objects. Complete the following steps:
```

----

+
*Note:* For installing the {acm-short} hub cluster on infrastructure nodes, see the xref:../install/install_connected.adoc#installing-on-infra-node[Installing the {acm-short} hub cluster on infrastructure nodes] section.

+
. Run the following command to apply the file and create the `MultiClusterHub` custom resource. Replace `custom-resource` with the name of your custom resource file:
 
+
[source,bash]
----
oc apply -f <path-to-file>/<custom-resource>.yaml
----

+

> If you receive the following error, the resource process is still running. Run the `oc apply` command again in a few minutes when the resources are created:

Reviewer 3: Reminder to add the [source,bash] syntax for the terminal message

- A `ServiceAccount` with a `ClusterRoleBinding` automatically gives cluster administrator privileges to {acm-short} and to any user credentials with access to the namespace where you install {acm-short}.

> - A namespace called `local-cluster` is reserved for the {product-title-short} hub cluster when it is self-managed. This is the only `local-cluster` namespace that can exist in the product. 

Reviewer 3: ```suggestion
- A namespace called `local-cluster` is reserved for the {acm-short} hub cluster when it is self-managed. This is the only `local-cluster` namespace that can exist in the product. 
```

      operator: Exists
----

. Add the following additional configuration before you apply the `MultiClusterHub` custom resource:

+
[source,yaml]
----
spec:
  nodeSelector:
    node-role.kubernetes.io/infra: ""
----

> 

Reviewer 3: I understand that this section was not apart of your changes, but I recommend that Add infrastructure nodes to the {ocp-short} cluster be changed to Adding infrastructure nodes to the {ocp-short} cluster

Reviewer 3: I also see that there is a link to Troubleshooting in the Installing from the {ocp-short} CLI section. If we want to keep it, let's move it to the Additional resources section

Author: Ok; I brought this up in the call one time recently as an example of what **not** to do. 

This is the work that this PR is holding up--rewriting this. 

We definitely don't even want to get in the habit of linking to troubleshooting at all, even in additional resources. Only in cases where we don't have a choice, but in this case, we got a few customer bugs, one requesting more troubleshooting, other requesting more info in the install doc. 

I am answering that by taking out the script and the link to t-shooting in an issue: https://issues.redhat.com/browse/ACM-14343

But I cannot start on it while this PR is still out. The link is going away, for sure. It should not have been placed here, and in another topic, the script is placed in the steps....so it is a but of a mess that I didn't write but inherited.

I will catch the infra node work in the next PR. Thanks for catching that.
